src/core/components/dht.js

```diff
@@ -167,7 +190,14 @@ module.exports = (self) => {
      * @param {function(Error, Array<PeerId>)} [callback]
      * @returns {Promise<Array<PeerId>>|void}
```

Comment:
We are using this type of returns with `Promise|void` in multiple other core components. We are probably using them because we wrap each function with `promisify`.

I haven't a strong opinion about if we should indicate a Promise return type if we use `promisify`, but if @alanshaw agrees with you, I can create a new PR for removing this type from all the core components defining a return Promise type in these cases.